cmd/geoipupdate/args.go

```diff
@@ -23,6 +24,13 @@ func getArgs() *Args {
 		confFileDefault = value
 	}
 
+	// Set the default config file only if it exists.
+	// Othwerwise, geoipupdate requires the user to specify the config file
+	// even if all the other arguments are set via the environment variables.
+	if _, err := os.Stat(confFileDefault); errors.Is(err, os.ErrNotExist) {
+		confFileDefault = ""
+	}
```

Comment:
Hmm, I wonder if this should go before the environment variable check. My thought is that if the environment variable is set, then you are explicitly specifying a configuration file and it should fail in that case.